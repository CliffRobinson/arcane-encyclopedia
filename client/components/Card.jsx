import React from "react";
import { Tooltip } from "react-tippy";

const Card = (props) => {
    let cardSize = "small";
    let hoverCardSize = "normal";

    switch (props.card.layout) {
    case "normal":
    case "saga":
    case "split":
        return (
            <React.Fragment> 
                <Tooltip
                    trigger='mouseenter'
                    html={<img src={props.card.image_uris[hoverCardSize]}></img>}
                >   
                    
                    <img src={props.card.image_uris[cardSize]}></img>
                    <button onClick={()=> console.log(props.card)}> Log this card </button>
                </Tooltip>
            </React.Fragment>
        );

    case "transform":
        return props.card.card_faces.map((face, i) => {
            return (
                <React.Fragment>
                    <Tooltip
                        trigger='mousenter'
                        html = {<img src={face.image_uris[hoverCardSize]}></img>}
                    >
                        
                        <img src={face.image_uris[cardSize]}></img>
                    </Tooltip>
                    <button onClick={()=> console.log(props.card)}> Log this card </button>
                </React.Fragment>
            );            
        });
    default:
        return (
            <div style={{maxWidth:"146 px", border:"1px solid black"}}>
                
                The card {props.card.name} has a weird format and we can't display it. 
                <button onClick={()=> console.log(props.card)}> Log this card </button>
            </div>
        );
        
    }
};

export default Card;