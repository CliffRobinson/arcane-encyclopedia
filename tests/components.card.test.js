import React from 'react';
import {shallow} from 'enzyme';

import Card from "../client/components/Card";
import {data as fakeCards} from "./testData";

test("regular card matches snapshot", ()=> {
    expect(
        toJson(shallow(<Card card={fakeCards[0]} />))
    ).toMatchSnapshot();
});

test("transform card matches snapshot", ()=>{
    expect(
        toJson(shallow(<Card card={fakeCards[7]} />))
    ).toMatchSnapshot();
});

test("not a card matches snapshot", ()=>{
    expect(
        toJson(shallow(<Card card={{}} />))
    ).toMatchSnapshot();
});
