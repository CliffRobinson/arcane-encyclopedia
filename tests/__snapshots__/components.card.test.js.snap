// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card matches snapshot 1`] = `
<React.Fragment>
  <Tooltip
    animateFill={true}
    animation="shift"
    arrow={false}
    arrowSize="regular"
    className=""
    delay={0}
    disabled={false}
    distance={10}
    duration={375}
    followCursor={false}
    hideDelay={0}
    hideDuration={375}
    hideOnClick={true}
    html={
      <img
        src="https://img.scryfall.com/cards/normal/en/rix/185.jpg?1524752669"
      />
    }
    inertia={false}
    interactive={false}
    interactiveBorder={2}
    multiple={false}
    offset={0}
    onHidden={[Function]}
    onHide={[Function]}
    onRequestClose={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    popperOptions={Object {}}
    position="top"
    size="regular"
    sticky={false}
    stickyDuration={200}
    style={Object {}}
    theme="dark"
    touchHold={false}
    trigger="mouseenter"
    unmountHTMLWhenHide={false}
  >
    <img
      src="https://img.scryfall.com/cards/small/en/rix/185.jpg?1524752669"
    />
  </Tooltip>
</React.Fragment>
`;

exports[`not a card matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid black",
      "maxWidth": "146 px",
    }
  }
>
  The card 
   has a weird format and we can't display it.
</div>
`;

exports[`regular card matches snapshot 1`] = `
<React.Fragment>
  <Tooltip
    animateFill={true}
    animation="shift"
    arrow={false}
    arrowSize="regular"
    className=""
    delay={0}
    disabled={false}
    distance={10}
    duration={375}
    followCursor={false}
    hideDelay={0}
    hideDuration={375}
    hideOnClick={true}
    html={
      <img
        src="https://img.scryfall.com/cards/normal/en/rix/185.jpg?1524752669"
      />
    }
    inertia={false}
    interactive={false}
    interactiveBorder={2}
    multiple={false}
    offset={0}
    onHidden={[Function]}
    onHide={[Function]}
    onRequestClose={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    popperOptions={Object {}}
    position="top"
    size="regular"
    sticky={false}
    stickyDuration={200}
    style={Object {}}
    theme="dark"
    touchHold={false}
    trigger="mouseenter"
    unmountHTMLWhenHide={false}
  >
    <img
      src="https://img.scryfall.com/cards/small/en/rix/185.jpg?1524752669"
    />
  </Tooltip>
</React.Fragment>
`;

exports[`transform card matches snapshot 1`] = `
Array [
  <React.Fragment />,
  <React.Fragment />,
]
`;
